core/utils/pcap_handle.cc

```diff
@@ -52,14 +52,18 @@ bool PcapHandle::is_initialized() const {
 }
 
 int PcapHandle::SendPacket(const u_char* packet, int len) {
-  if (len <= PCAP_SNAPLEN) {
+  if (is_initialized() && len <= PCAP_SNAPLEN) {
```

Comment:
Minor question -- I think this is fine here, but perhaps in the higher performance drivers we should expect the caller to check if it's initialized before making any send/receive calls, so that this check doesn't have to happen per packet.